compiler-plugin/src/main/kotlin/com/google/devtools/ksp/symbol/impl/binary/KSAnnotationDescriptorImpl.kt

```diff
@@ -113,7 +113,8 @@ private fun <T> ConstantValue<T>.toValue(parent: KSNode): Any? = when (this) {
             it.simpleName.asString() == value.second.asString()
     }?.let { (it as KSClassDeclaration).asStarProjectedType() }
     is KClassValue -> when (val classValue = value) {
-        is KClassValue.Value.NormalClass -> classValue.classId.findKSType()
+        is KClassValue.Value.NormalClass -> if (classValue.arrayDimensions > 0)
+            ResolverImpl.instance.builtIns.arrayType else classValue.classId.findKSType()
```

Comment:
`builtIns.arrayType` is `Array<*>`. We probably need `arrayType.replace(listOf(ResolverImpl.instance.getTypeArgument(classValue.classId.findKSType(), Variance.INVARIANT)))`.

Reply:
It is not possible to do `Array<T>.class` in Java, and it is only possible in Kotlin with `Array` as the type, therefore I decided to put just `Array` here for consistence.